Stratis.Bitcoin/BlockStore/BlockRepository.cs

```diff
@@ -293,40 +453,203 @@ public Task DeleteAsync(uint256 newlockHash, List<uint256> hashes)
+                                    var key2 = input.PrevOut.Hash.ToBytes().Concat(input.PrevOut.N.ToBytes());
+                                    this.PerformanceCounter.AddRepositoryDeleteCount(1);
+                                    this.session.Transaction.RemoveKey<byte[]>("Output", key2);
+                                }
                             }
+
+                            foreach (KeyValuePair<uint160, HashSet<byte[]>> kv in removals)
+                                this.PerformanceCounter.AddRepositoryDeleteCount(
+                                    this.RemoveTransactionOutputsToAddress(kv.Key, kv.Value));
                         }
-                        else {
+                        else
+                        {
                             this.PerformanceCounter.AddRepositoryMissCount(1);
                         }
-					}
+			        }
 
                     this.PerformanceCounter.AddRepositoryDeleteCount(1);
                     this.session.Transaction.RemoveKey<byte[]>("Block", key);
-				}
+			    }
 
-				this.SaveBlockHash(newlockHash);
-				this.session.Transaction.Commit();
+			    this.SaveBlockHash(newlockHash);
+			    this.session.Transaction.Commit();
 			});
 		}
 
+        private IEnumerable<byte[]> GetAddressTransactionOutputs(byte[] addr)
+        {
+            var addrRow = this.session.Transaction.Select<byte[], byte[]>("Address", addr);
+            if (addrRow.Exists)
+            {
+                // Get the number of transactions
+                var cntBytes = addrRow.GetValuePart(0, sizeof(uint));
+                if (!BitConverter.IsLittleEndian) Array.Reverse(cntBytes);
+                uint cnt = BitConverter.ToUInt32(cntBytes, 0);
+
+                // Any transactions recorded?
+                if (cnt > 0)
+                {
+                    const int elementSize = 36;
+
+                    byte[] listBytes = addrRow.GetValuePart(sizeof(uint), (uint)(cnt * elementSize));
+                    for (int i = 0; i < cnt; i++)
+                    {
+                        var trxid = new byte[32];
+                        Array.Copy(listBytes, i * elementSize, trxid, 0, elementSize);
+                        yield return trxid;
+                    }
+                }
+            }
+        }
+
+        private void RecordTransactionOutputsToAddress(uint160 addr, HashSet<byte[]> list)
+        {
+            uint cnt = 0;
+            var addrRow = this.session.Transaction.Select<byte[], byte[]>("Address", addr.ToBytes());
+
+            if (addrRow.Exists)
+            {
+                // Get the number of transaction hashes
+                var bytes = addrRow.GetValuePart(0, sizeof(uint));
+                if (!BitConverter.IsLittleEndian) Array.Reverse(bytes);
+                cnt = BitConverter.ToUInt32(bytes, 0);
+            }
+
+            const int elementSize = 36; // Transaction Hash + output number = sizeof(trxid) + sizeof(uint)
```

Comment:
If this is only a constant local to the method I would like to see it at the top of the method. If it's used elsewhere in the class put it above the constructor near the performancecounter.